File: 2056_연월일달력.cpp
#include<iostream>
#include <string>

using namespace std;

int main(int argc, char** argv)
{
	int test_case;
	int T;
	string n;
	cin >> T;
	for (test_case = 1; test_case <= T; ++test_case)
	{
		cin >> n;
		int m = stoi(n);
		int mon = (m % 10000) / 100;
		int day = m % 100;
		cout << "#" << test_case << " ";
		if (day < 1 or mon < 1)
		{
			cout << "-1\n";
			continue;
		}
		if (mon == 2)
		{
			if (day <= 28)
			{
				n.insert(4, "/");
				n.insert(7, "/");
				cout << n << "\n";
				continue;
			}
		}
		else if (mon <= 7)
		{
			if (mon % 2 == 0)
			{
				if (day <= 30)
				{
					n.insert(4, "/");
					n.insert(7, "/");
					cout << n << "\n";
					continue;
				}
			}
			else
			{
				if (day <= 31)
				{
					n.insert(4, "/");
					n.insert(7, "/");
					cout << n << "\n";
					continue;
				}
			}
		}
		else
		{
			if (mon % 2 == 1)
			{
				if (day <= 30)
				{
					n.insert(4, "/");
					n.insert(7, "/");
					cout << n << "\n";
					continue;
				}
			}
			else
			{
				if (day <= 31)
				{
					n.insert(4, "/");
					n.insert(7, "/");
					cout << n << "\n";
					continue;
				}
			}
		}
		cout << "-1\n";
	}
	return 0;
}

File: 2056_연월일달력.py
T = int(input())
for test_case in range(1, T + 1):
    n = input()
    month = int(n[4:6])
    day = int(n[6:])
    print("#%d " %(test_case), end='')
    if day <= 0:
        print("-1")
        continue
    if month == 2:
        if day <= 28:
            print(f"{n[:4]}/{n[4:6]}/{n[6:]}")
            continue
    elif month in range(1, 8, 2) or month in range(8, 13, 2):
        if day <= 31:
            print(f"{n[:4]}/{n[4:6]}/{n[6:]}")
            continue
    elif month in range(2, 7, 2) or month in range(9, 12, 2):
        if day <= 30:
            print(f"{n[:4]}/{n[4:6]}/{n[6:]}")
            continue
    print("-1")